processing/stenosis_connector.go

```diff
@@ -86,7 +86,7 @@ func MakeStenosisConnector(endpoint string, timeout, timeBracket time.Duration,
 			log.Debugf("flow with existing alert finished: %v", flow)
 			myAlerts = aval.([]types.Entry)
 			outParsed, err := sConn.submit(&flow)
-			if err != nil {
+			if err != nil || outParsed == nil {
```

Comment:
Since this OR would allow `err` to be nil, the `log.Error` would simply log `<nil>`, which is not nice.
Maybe adapt the log.Error saying "nothing got parsed" when err is nil?

Reply:
On the other hand, if `sConn.submit` returns an error, Go convention would be to return result as nil (when it is a pointer value). Can this ever happen? Should this ever happen?

Reply:
The intention here is to be as robust as possible and to never dereference anything that could be `nil`. Here, when `sConn.submit()` returns an error, I just want to log a message and get the alerts out nevertheless (I wouldn't even know where to "return" `nil` to in this case, as this loop is a background job that gets started when creating the connector). Here, the problem is that sometimes apparently `err` is not `nil` but `outParsed` is. I can't see how that could happen in the code, we have seen a rare segfault based on that in the wild. Given that the call chain leading from `sConn.submit` disappears somewhere in the gRPC code, I didn't want to go all the way down the rabbit hole. This is just to make sure we don't *crash* FEVER by dereferencing something wrong.

I'll make sure that some sensible error message is logged though. This makes it easier to debug this case in more detail in case the problem appears again.